okhttp/src/main/java/okhttp3/internal/http/RetryAndFollowUpInterceptor.java

```diff
@@ -164,13 +169,14 @@ public StreamAllocation streamAllocation() {
       }
 
       if (followUp.body() instanceof UnrepeatableRequestBody) {
+        streamAllocation.release();
```

Comment:
The response body is getting closed a few lines up. I'll look further into this to make sure.